src/selectors/cashTransaction.js

```diff
@@ -0,0 +1,110 @@
+
+const selectAmount = createSelector([selectTransaction], ({ amount }) => amount);
+
+const selectPaymentType = createSelector([selectTransaction], ({ paymentType }) => paymentType);
+
+const selectReason = createSelector([selectTransaction], ({ reason }) => reason);
+
+const selectDescription = createSelector([selectTransaction], ({ description }) => description);
+
+const selectBalance = createSelector([selectPaymentType], paymentType => {
+  const receiptBalance = UIDatabase.objects('Transaction')
+    .filtered('type == "receipt" && paymentType.code == $0', paymentType?.code)
+    .sum('subtotal');
+  const paymentBalance = UIDatabase.objects('Transaction')
+    .filtered('type == "payment" && paymentType.code == $0', paymentType?.code)
+    .sum('subtotal');
+  return currency(receiptBalance - paymentBalance);
+});
+
+const selectAmountBuffer = createSelector([selectTransaction], ({ amountBuffer }) => amountBuffer);
+
+const selectDescriptionBuffer = createSelector(
+  [selectTransaction],
+  ({ descriptionBuffer }) => descriptionBuffer
+);
+
+const selectInputModal = createSelector([selectTransaction], ({ inputModal }) => inputModal);
+
+const selectIsInputModalOpen = createSelector([selectInputModal], ({ isOpen }) => isOpen);
+
+const selectInputModalField = createSelector([selectInputModal], ({ field }) => field);
+
+const selectIsInputNameModalOpen = createSelector(
+  [selectIsInputModalOpen, selectInputModalField],
+  (isInputModalOpen, inputModalField) =>
+    isInputModalOpen && inputModalField === CASH_TRANSACTION_INPUT_MODAL_FIELDS.NAME
+);
+
+const selectIsInputAmountModalOpen = createSelector(
+  [selectIsInputModalOpen, selectInputModalField],
+  (isInputModalOpen, inputModalField) =>
+    isInputModalOpen && inputModalField === CASH_TRANSACTION_INPUT_MODAL_FIELDS.AMOUNT
+);
+
+const selectIsInputPaymentTypeModalOpen = createSelector(
+  [selectIsInputModalOpen, selectInputModalField],
+  (isInputModalOpen, inputModalField) =>
+    isInputModalOpen && inputModalField === CASH_TRANSACTION_INPUT_MODAL_FIELDS.PAYMENT_TYPE
+);
+
+const selectIsInputReasonModalOpen = createSelector(
+  [selectIsInputModalOpen, selectInputModalField],
+  (isInputModalOpen, inputModalField) =>
+    isInputModalOpen && inputModalField === CASH_TRANSACTION_INPUT_MODAL_FIELDS.REASON
+);
+
+const selectIsInputDescriptionModalOpen = createSelector(
+  [selectIsInputModalOpen, selectInputModalField],
+  (isInputModalOpen, inputModalField) =>
+    isInputModalOpen && inputModalField === CASH_TRANSACTION_INPUT_MODAL_FIELDS.DESCRIPTION
+);
+
+const selectIsValid = createSelector(
+  [selectName, selectType, selectAmount, selectPaymentType, selectReason],
+  (name, type, amount, paymentType, reason) =>
+    !!name &&
+    !!amount &&
+    amount > currency(0) &&
+    (type === CASH_TRANSACTION_TYPES.CASH_IN || !!reason) &&
+    paymentType
+);
+
+export const CashTransactionSelectors = {
```

Comment:
Put this comment in the wrong place!

Tried something a bit different here, felt like it fit nicely with how we scope actions and reducers, e.g.

```
src/
  actions/
    index.js
    CashTransactionActions.js
  reducers/
    index.js
    CashTransactionReducer.js
  selectors/
    index.js
    CashTransactionSelectors.js
```